spec/controllers/application_controller_spec.rb

```diff
@@ -1,114 +1,57 @@
-          .and_return("255.255.255.255")
-        expect(controller.request_ip).to eql("255.255.255.0")
-      end
+  describe "page_visited events" do
+    it "triggers a `page_visited` event on a request" do
+      expect { get :test_action }.to have_enqueued_job(SendEventToDataWarehouseJob).with(
+        "events",
+        hash_including(type: :page_visited, request_path: "/test_action"),
+      )
     end
   end
 
   describe "sets headers" do
     it "robots are asked not to index or to follow" do
-      get :check
+      get :test_action
       expect(response.headers["X-Robots-Tag"]).to eq("noindex, nofollow")
     end
   end
 
   describe "#strip_empty_checkboxes" do
-    context "when the checkbox value is an Array" do
-      let(:params) do
-        { test_form: { test_field: checkboxes } }
-      end
-
-      before do
-        allow(controller).to receive(:params).and_return(params)
-      end
-
-      context "no empty checkboxes added by GOVUKDesignSystemFormBuilder" do
-        let(:checkboxes) { %w[first_box second_box] }
+    controller do
+      before_action { strip_empty_checkboxes(%i[nothing_to_remove_field array_field string_field], :test_form) }
 
-        it "removes nothing from the array" do
-          subject.strip_empty_checkboxes(%i[test_field], :test_form)
-          expect(controller.params[:test_form][:test_field]).to eql(checkboxes)
-        end
-      end
-
-      context "empty checkbox added by GOVUKDesignSystemFormBuilder" do
-        let(:checkboxes) { ["first_box", "second_box", ""] }
-        let(:stripped_checkboxes) { %w[first_box second_box] }
-
-        it "removes empty checkbox from the array" do
-          subject.strip_empty_checkboxes(%i[test_field], :test_form)
-          expect(controller.params[:test_form][:test_field]).to eql(stripped_checkboxes)
-        end
+      def strip_test_action
+        head :ok
       end
     end
 
-    context "when the checkbox value is a String" do
-      let(:params) do
-        { test_form: { array_field: ["first_box", ""], string_field: "" } }
-      end
-
-      before do
-        allow(controller).to receive(:params).and_return(params)
-      end
+    before do
+      routes.draw { get "strip_test_action" => "anonymous#strip_test_action" }
+
+      get :strip_test_action, params: {
+        test_form: {
+          nothing_to_remove_field: %w[first_box second_box],
+          array_field: ["first_box", ""],
+          string_field: "",
+        },
+      }
+    end
 
-      it "removes empty checkbox from the array without error" do
-        subject.strip_empty_checkboxes(%i[array_field string_field], :test_form)
-        expect(controller.params[:test_form][:array_field]).to eql(%w[first_box])
-        expect(controller.params[:test_form][:string_field]).to eql("")
-      end
+    it "removes empty checkboxes as expected" do
+      expect(controller.params[:test_form][:nothing_to_remove_field]).to eql(%w[first_box second_box])
```

Comment:
I think I'm guilty of more than a few of these, but do we want to move towards `eq` across the board unless we explicitly want to check a bit more strictly?

Reply:
Ah I didn't even spot that – it was on my list of little Firebreak things to do 🚒 

I'll leave it in this PR and do that next!